flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java

```diff
@@ -172,10 +172,16 @@
 
 	/**
 	 * Descriptor of the transactional IDs list.
+	 * Note: This state is based on the Java serializer and it has compatibility problem that will be removed later.
```

Comment:
Are you sure that the serialization was done by Java?

From my speculation on the `NextTransactionalIdHint`, it should have been serialized by the `KryoSerializer`.

Reply:
It seems you are right. Will change the note.